src/types.test.ts

```diff
@@ -1003,6 +851,30 @@ describe('c.var with chaining - test only types', () => {
 
       return c.json(0)
     })
+
+    type Env = {
+      Variables: {
+        init: number
+      }
+    }
+
+    new Hono<Env>()
+      .get('/', mw1, (c) => {
+        expectTypeOf(c.get('init')).toEqualTypeOf<number>()
+        expectTypeOf(c.var.init).toEqualTypeOf<number>()
+        expectTypeOf(c.get('foo1')).toEqualTypeOf<string>()
+        expectTypeOf(c.var.foo1).toEqualTypeOf<string>()
+        return c.json(0)
+      })
+      .get('/', (c) => {
+        expectTypeOf(c.get('init')).toEqualTypeOf<number>()
+        expectTypeOf(c.var.init).toEqualTypeOf<number>()
+        // @ts-expect-error foo1 is not typed
+        c.get('foo1')
+        // @ts-expect-error foo1 is not typed
+        c.var.foo1
+        return c.json(0)
+      })
```

Comment:
If you can, I'd like you to write the rest of the test too.

```ts
new Hono<Env>()
.get('/', mw1, mw2, (c) => {
  expectTypeOf(c.get('init')).toEqualTypeOf<number>()
  expectTypeOf(c.var.init).toEqualTypeOf<number>()
  expectTypeOf(c.get('foo1')).toEqualTypeOf<string>()
  expectTypeOf(c.var.foo1).toEqualTypeOf<string>()
  expectTypeOf(c.get('foo2')).toEqualTypeOf<string>()
  expectTypeOf(c.var.foo2).toEqualTypeOf<string>()
  return c.json(0)
})
.get('/', (c) => {
  expectTypeOf(c.get('init')).toEqualTypeOf<number>()
  expectTypeOf(c.var.init).toEqualTypeOf<number>()
  // @ts-expect-error foo1 is not typed
  c.get('foo1')
  // @ts-expect-error foo1 is not typed
  c.var.foo1
  // @ts-expect-error foo1 is not typed
  c.get('foo2')
  // @ts-expect-error foo1 is not typed
  c.var.foo2
  return c.json(0)
})

// ...
```

Reply:
Sorry for late reply. I'll try it.

Reply:
@yusukebe I added tests for `app.get(path, ...handlers x 3)` ~ `app.get(path, ...handlers x 10)`!